.ci/magician/exec/runner.go

```diff
@@ -100,6 +100,14 @@ func (ar *Runner) ReadFile(name string) (string, error) {
 	return string(data), nil
 }
 
+func (ar *Runner) AppendFile(name, data string) error {
```

Comment:
Rather than reading & then writing the file, we may want to use append modes - more efficient: https://pkg.go.dev/os#example-OpenFile-Append

Reply:
Does this look correct to you?

This function doesn't get called yet, so I'll save testing it for the next PR.

Reply:
That looks about right; I'm not super familiar with doing this though.